lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -492,6 +470,58 @@ void ExportVerilog::prepareHWModule(Block &block,
       lowerAlwaysInlineOperation(&op);
       continue;
     }
+
+    // If this expression is deemed worth spilling into a wire, do it here.
+    if (shouldSpillWire(op, options)) {
+      // If we're not in a procedural region, or we are, but we can hoist out of
+      // it, we are good to generate a wire.
+      if (!isProceduralRegion ||
+          (isProceduralRegion && hoistNonSideEffectExpr(&op)))
+        lowerUsersToTemporaryWire(op);
+
+      // If we're in a procedural region, we move on to the next op in the
+      // block. The expression splitting and canonicalization below will happen
+      // after we recurse back up. If we're not in a procedural region, the
+      // expression can continue being worked on.
+      if (isProceduralRegion) {
+        ++opIterator;
+        continue;
+      }
+    }
+
+    // Lower variadic fully-associative operations with more than two operands
```

Comment:
this comment isn't correct anymore.

Reply:
This part didn't change, it still makes a binary tree as before. But it's doing so after potentially spilling this expression into a wire assignment.